dspace-uclouvain/src/main/java/org/dspace/uclouvain/authority/UCLouvainPromoterAuthority.java

```diff
@@ -0,0 +1,82 @@
+package org.dspace.uclouvain.authority;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.dspace.content.authority.Choice;
+import org.dspace.content.authority.ChoiceAuthority;
+import org.dspace.content.authority.Choices;
+import org.dspace.uclouvain.authority.client.UCLouvainAuthorityClient;
+import org.dspace.uclouvain.authority.configuration.UCLouvainAuthorAuthorityAPIConfiguration;
+import org.dspace.uclouvain.authority.factory.UCLouvainServiceFactory;
+import org.dspace.uclouvain.external.dilbert.model.DialPerson;
+
+public class UCLouvainPromoterAuthority implements ChoiceAuthority {
+    
+    private String pluginInstanceName;
+
+    private UCLouvainAuthorAuthorityAPIConfiguration getUCLouvainAuthorAuthorityConfiguration() {
+        return UCLouvainServiceFactory.getInstance().getUCLouvainAuthorAuthorityConfiguration();
+    }
+
+    private UCLouvainAuthorityClient getUCLouvainAuthorityClient() {
+        return UCLouvainServiceFactory.getInstance().getUCLouvainAuthorityClient();
+    }
+
+    @Override
+    public Choices getMatches(String query, int start, int limit, String locale) {
+        // Implement authority for promoters
+        DialPerson[] persons = getPromoterList(query);
```

Comment:
The `getSudentList()` (UCLouvainAuthorAuthority class), `getPromoterList()` must be abstracted into a generic `getSuggestions()` method. This method should be declare either into an interface, either into the abstract super class.

Doing, this job, this entire method could be placed into the super AbstractClass